src/main/java/core/basesyntax/services/actions/PurchaseActionHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.services.actions;
+
+import core.basesyntax.db.Storage;
+
+public class PurchaseActionHandler implements ActionHandler {
+    private static Storage fruitDB;
+
+    public PurchaseActionHandler(Storage fruitDB) {
+        this.fruitDB = fruitDB;
+    }
+
+    @Override
+    public boolean actionStore(String nameOfGoods, Integer valueOfTask) {
+        validData(fruitDB, nameOfGoods);
+        Integer value = fruitDB.getStorageFruits().get(nameOfGoods);
+        return fruitDB.add(nameOfGoods, value - valueOfTask);

```

Comment:
what happens if I try to buy 100 bananas when storage contains only 5?

Reply:
not resolved